lib/magpie_web/router.ex

```diff
@@ -1,8 +1,14 @@
 defmodule Magpie.Router do
   use MagpieWeb, :router
   use Plug.ErrorHandler
+  import Plug.BasicAuth
 
   pipeline :browser do
+    if Application.get_env(:magpie, :no_basic_auth) == false do
+      username = Application.get_env(:magpie, :authentication)[:username]
```

Comment:
FYI @marcelklehr I think this actually didn't work when deploying the app afresh, since the Router module is compiled at compile-time, but we previously fetched the username and password in `releases.exs`, which is evaluated at runtime. This was probably the reason the logic didn't live here before (I also mentioned the distinction between compile time and runtime in the discussion of this MR AFAIK). This should be fixed in the latest master today.